Python/thread_pthread.h

```diff
@@ -302,6 +308,26 @@ PyThread_get_thread_ident(void)
     return (unsigned long) threadid;
 }
 
+#ifdef PY_HAVE_THREAD_NATIVE_ID
+unsigned long
+PyThread_get_thread_native_id(void)
+{
+    if (!initialized)
+        PyThread_init_thread();
+#ifdef __APPLE__
+    uint64_t native_id;
+    pthread_threadid_np(NULL, &native_id);
+#elif defined(__linux__)
+    pid_t native_id;
+    native_id = syscall(__NR_gettid);
+#elif defined(__FreeBSD__)
+    pid_t native_id;
+    native_id = pthread_getthreadid_np();
```

Comment:
Are you sure about the "pid_t" type? The following FreeBSD manual page documents the return type as "int":
https://www.freebsd.org/cgi/man.cgi?query=pthread_getthreadid_np&sektion=3&manpath=freebsd-release-ports

Reply:
Quite right – I think this was left over from the Linux implementation. Fixed :+1: